.github/workflows/prettier-on-push.yml

```diff
@@ -0,0 +1,38 @@
+name: Prettier code formatter (Push)
+
+on:
+  push:
+    branches:
+      - master
+      - main
+  workflow_dispatch:
+
+jobs:
+  check:
+    # available images: https://github.com/actions/runner-images#available-images
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
+      - name: Setup Node.js ⚙️
+        uses: actions/setup-node@v4
+      - name: Install Prettier 💾
+        run: npm install --save-dev --save-exact prettier @shopify/prettier-plugin-liquid
+      - name: Prettier Check 🔎
+        id: prettier
+        run: npx prettier . --check
+      - name: Show diff 📝
+        # https://docs.github.com/en/actions/learn-github-actions/expressions#failure
+        if: ${{ failure() }}
+        run: |
+          npx prettier . --write
+          git diff -- . ':(exclude)package-lock.json' ':(exclude)package.json' > diff.txt
+          npm install -g diff2html-cli
+          diff2html -i file -s side -F diff.html -- diff.txt
+      - name: Upload html diff
+        if: ${{ failure() && steps.prettier.conclusion == 'failure' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: HTML Diff
+          path: diff.html
+          retention-days: 3
```

Comment:
I'm not sure if we need this artifact generation on pushes. most of the contributions are done through PRs anyway and prettier check runs on PRs. I can also protect the master branch from direct pushes, making PRs the only way to submit changes. what do you think?

Reply:
It can be a good idea to protect the master branch, but independently I think uploading this artifact is useful.

For example, sometimes you want to fix small things and then end up doing this from the web interface instead of from local setup. For these cases having a report of what are the changes that you need to do instead of having to "guess" or run from local setup is useful.

Also a lot of our users are not from coding background and might not have installed prettier in their setup, so having a way to check for errors is nice. For this matter it would be useful to also have prettier installed as a local dependency in the template, and somehow setup to be run before submitting the PRs, but idk how to do that.